assets/js/blocks/products/all-products/index.js

```diff
@@ -24,86 +25,9 @@ export const blockSettings = {
 			/>
 		),
 	},
-	category: 'woocommerce',
-	keywords: [ __( 'WooCommerce', 'woo-gutenberg-products-block' ) ],
-	description: __(
-		'Display products from your store in a grid layout.',
-		'woo-gutenberg-products-block'
-	),
-	supports: {
-		align: [ 'wide', 'full' ],
-		html: false,
-		multiple: false,
-	},
-	example: {
-		attributes: {
-			isPreview: true,
-		},
-	},
-	attributes: sharedAttributes,
-	defaults,
-	/**
-	 * Renders and manages the block.
-	 *
-	 * @param {Object} props Props to pass to block.
-	 */
-	edit( props ) {
-		return <Editor { ...props } />;
-	},
+	edit,
 	// Save the props to post content.
-	save( { attributes } ) {
-		const dataAttributes = {};
-		Object.keys( attributes )
-			.sort()
-			.forEach( ( key ) => {
-				dataAttributes[ key ] = attributes[ key ];
-			} );
-		const data = {
-			'data-attributes': JSON.stringify( dataAttributes ),
-		};
-		return (
-			<div
-				className={ getBlockClassName(
-					'wc-block-all-products',
-					attributes
-				) }
-				{ ...data }
-			>
-				<InnerBlocks.Content />
-			</div>
-		);
-	},
+	save,
+	deprecated,
+	defaults,
 };
-
-/**
- * Register and run the "All Products" block.
- */
-registerBlockType( 'woocommerce/all-products', {
-	...blockSettings,
-	/**
-	 * Deprecation rule to handle the previous default rows which was 1 instead of 3.
-	 */
-	deprecated: [
-		{
-			attributes: Object.assign( {}, blockSettings.attributes, {
-				rows: { type: 'number', default: 1 },
-			} ),
-			save( { attributes } ) {
-				const data = {
-					'data-attributes': JSON.stringify( attributes ),
-				};
-				return (
-					<div
-						className={ getBlockClassName(
-							'wc-block-all-products',
-							attributes
-						) }
-						{ ...data }
-					>
-						<InnerBlocks.Content />
-					</div>
-				);
-			},
-		},
-	],
-} );
```

Comment:
I believe that this is no longer needed -- see PR description:

> ( [...] rely on server-side registration via `register_block_type_from_metadata` [here](https://github.com/woocommerce/woocommerce-blocks/blob/22bbecfeb4f4176dbeaeeab67d18f429d3d43656/src/BlockTypes/AbstractBlock.php#L213-L216)) thanks to the existing [`AllProducts.php` file](https://github.com/woocommerce/woocommerce-blocks/blob/22bbecfeb4f4176dbeaeeab67d18f429d3d43656/src/BlockTypes/AllProducts.php) and class

Reply:
You still need to register client-specific settings with `registerBlockType`. I don't think you need to export anything from this file. Depending on the lowest supported WP version you only need to call:

```js
import metadata from './block.json';
registerBlockType( metadata.name, settings );
```

or

```js
import metadata from './block.json';
registerBlockType( metadata, settings );
```

In case some settings aren't handled in older versions of WP.

Reply:
Thanks @gziolo! Addressed in 399e3b0.